podcasts/DownloadManager+URLSessionDelegate.swift

```diff
@@ -36,19 +36,24 @@ extension DownloadManager: URLSessionDelegate, URLSessionDownloadDelegate {
             progressManager.updateProgressForEpisode(downloadingEpisode.uuid, totalBytesWritten: totalBytesWritten, totalBytesExpected: totalBytesExpectedToWrite)
         }
 
-        if !downloadingEpisode.downloading(), downloadingEpisode.downloadTaskId != nil {
+        // If our download status or downloadTaskId are incorrect, then we should update these
+        if !downloadingEpisode.downloading() || downloadingEpisode.downloadTaskId == nil {
```

Comment:
I was finding that `downloadTaskId` was missing from some downloads, causing the download progress to fail. This didn't seem to be new with the changes so I think it makes sense to just set this value when we update the status to `downloading`.